src/Shared/FileUtilities.cs

```diff
@@ -315,9 +315,14 @@ private static bool IsValidRelativePathBound(char? c)
         /// </summary>
         internal static string NormalizePath(string path)
         {
-
             ErrorUtilities.VerifyThrowArgumentLength(path, nameof(path));
+
+#if CLR2COMPATIBILITY
             string fullPath = GetFullPath(path);
+#else
+            // Windows uses a case-insensitive file system, so we need to get an exact file name from a possibly-differently-cased path.
+            string fullPath = NativeMethodsShared.IsWindows ? GetExactPath(path) : GetFullPath(path);
```

Comment:
FIY, building OrchardCore invokes this method ~47000 times.

Reply:
~100,000 times on WebLargeCore